src/servers/http_server_v2.cc

```diff
@@ -0,0 +1,1539 @@
+               ? EVHTP_RES_OK
+               : EVHTP_RES_BADREQ);
+
+  TRTSERVER_ErrorDelete(err);
+}
+
+void
+HTTPAPIServerV2::HandleServerMetadata(evhtp_request_t* req)
+{
+  if (req->method != htp_method_GET) {
+    evhtp_send_reply(req, EVHTP_RES_METHNALLOWED);
+    return;
+  }
+
+  rapidjson::Document document;
+  document.SetObject();
+  rapidjson::Document::AllocatorType& allocator = document.GetAllocator();
+  const char* name = nullptr;
+  TRTSERVER_Error* err = TRTSERVER_ServerId(server_.get(), &name);
+  if (err == nullptr) {
+    std::string name_str(name);
+    rapidjson::Value name_val(name_str.c_str(), name_str.size());
+    document.AddMember("name", name_val, allocator);
+
+    const char* version = nullptr;
+    err = TRTSERVER_ServerVersion(server_.get(), &version);
+    if (err == nullptr) {
+      std::string version_str(version);
+      rapidjson::Value version_val(version_str.c_str(), version_str.size());
+      document.AddMember("version", version_val, allocator);
+
+      uint64_t extensions_count;
+      const char* const* extensions;
+      err = TRTSERVER_ServerExtensions(
+          server_.get(), &extensions, &extensions_count);
+      rapidjson::Value extensions_array(rapidjson::kArrayType);
+      if (err == nullptr) {
+        for (uint64_t i = 0; i < extensions_count; ++i) {
+          std::string extension_str(extensions[i]);
+          rapidjson::Value extension_val(
+              extension_str.c_str(), extension_str.size(), allocator);
+          extensions_array.PushBack(extension_val, allocator);
+        }
+        document.AddMember("extensions", extensions_array, allocator);
+      }
+    }
+  }
+
+  if (err == nullptr) {
+    rapidjson::StringBuffer buffer;
+    buffer.Clear();
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    document.Accept(writer);
+    std::string status_buffer(buffer.GetString());
+
+    evbuffer_add(req->buffer_out, status_buffer.c_str(), status_buffer.size());
+  }
+
+  evhtp_headers_add_header(
+      req->headers_out,
+      evhtp_header_new("Content-Type", "application/json", 1, 1));
+
+  RequestStatus request_status;
+  RequestStatusUtil::Create(
+      &request_status, err, RequestStatusUtil::NextUniqueRequestId(),
+      server_id_);
+
+  evhtp_headers_add_header(
+      req->headers_out,
+      evhtp_header_new(
+          kStatusHTTPHeader, request_status.ShortDebugString().c_str(), 1, 1));
+
+  evhtp_send_reply(
+      req, (request_status.code() == RequestStatusCode::SUCCESS)
+               ? EVHTP_RES_OK
+               : EVHTP_RES_BADREQ);
+
+  TRTSERVER_ErrorDelete(err);
+}
+
+TRTSERVER_Error*
+HTTPAPIServerV2::EVBufferToInput(
+    const std::string& model_name, const InferRequestHeader& request_header,
+    evbuffer* input_buffer,
+    TRTSERVER_InferenceRequestProvider* request_provider,
+    AllocPayload* alloc_payload)
+{
+  // Extract individual input data from HTTP body and register in
```

Comment:
Why commented out? Again you likely want to base this file off http_server.cc as that is the up-to-date code.

Reply:
Fixed.